src/SmartTransactionsController.ts

```diff
@@ -889,4 +889,56 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       );
     });
   }
+
+  #wipeSmartTransactionsPerChainId(
+    chainId: Hex,
+    selectedAddressLowerCase: string,
+  ): void {
+    const { smartTransactions } = this.state.smartTransactionsState;
+    const smartTransactionsForSelectedChain: SmartTransaction[] =
+      smartTransactions?.[chainId];
+    if (
+      !smartTransactionsForSelectedChain ||
+      smartTransactionsForSelectedChain.length === 0
+    ) {
+      return;
+    }
+    const newSmartTransactionsForSelectedChain =
+      smartTransactionsForSelectedChain.filter(
+        (smartTransaction: SmartTransaction) =>
+          smartTransaction.txParams?.from !== selectedAddressLowerCase,
+      );
+    this.update({
+      smartTransactionsState: {
+        ...this.state.smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactions,
+          [chainId]: newSmartTransactionsForSelectedChain,
+        },
+      },
+    });
+  }
+
+  wipeSmartTransactions(
+    selectedAddress?: string,
+    ignoreNetwork?: boolean,
+  ): void {
+    if (!selectedAddress) {
```

Comment:
Is this intentional? In the `TransactionController`, the `selectedAddress` is optional since it's an optional filter meaning if it's not provided, transactions for all accounts are removed.

Reply:
Yes, I have this check and early return as part of defensive programming, just in case the value would be falsy (e.g. an empty string), since this function will be called from a .js file, which is dynamically typed. However, I will make a tiny change to make this param mandatory.